src/main/java/com/liveramp/daemon_lib/builders/BaseForkingDaemonBuilder.java

```diff
@@ -0,0 +1,70 @@
+package com.liveramp.daemon_lib.builders;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+
+import com.google.common.collect.Maps;
+import org.jetbrains.annotations.NotNull;
+
+import com.liveramp.daemon_lib.JobletCallback;
+import com.liveramp.daemon_lib.JobletConfig;
+import com.liveramp.daemon_lib.JobletConfigProducer;
+import com.liveramp.daemon_lib.JobletFactory;
+import com.liveramp.daemon_lib.executors.JobletExecutor;
+import com.liveramp.daemon_lib.executors.JobletExecutors;
+import com.liveramp.daemon_lib.executors.forking.ProcessJobletRunner;
+
+public abstract class BaseForkingDaemonBuilder<T extends JobletConfig, E extends BaseForkingDaemonBuilder<T, E>> extends BaseDaemonBuilder<T, E> {
+
+  private final String workingDir;
+  private final Class<? extends JobletFactory<T>> jobletFactoryClass;
```

Comment:
Another thing we could do is put the jobletFactory in the BaseDaemonBuilder, and call .getClass() on it here.  This would be a fairly annoying breaking change though, so it's probably not worth it.